Add documentation for using Python packages (#966)
ros2/ros2_documentation: source/Tutorials/Using-Python-Packages.rst

Installing via ``rosdep``
-------------------------

The fastest way to include third-party python packages is to use their corresponding rosdep keys, if available.  ``rosdep`` keys can be checked via:

* https://github.com/ros/rosdistro/blob/master/rosdep/base.yaml
* https://github.com/ros/rosdistro/blob/master/rosdep/python.yaml

These ``rosdep`` keys can be added to your ``package.xml`` file, which indicates to the build system that your package (and dependent packages) depend on those keys. In a new workspace, you can also quickly install all rosdep keys with:

.. code-block:: console

    rosdep install -yr ./path/to/your/workspace

If there aren’t currently ``rosdep`` keys for the package that you are interested in, it is possible to add them by following the `rosdep key contribution guide`_.

> 

Reviewer: ```suggestion

To learn more about the ``rosdep`` tool and how it works, consult the `rosdep documentation`_.
```

Reviewer: And add reference: http://docs.ros.org/en/independent/api/rosdep/html/

Installing via a package manager
--------------------------------

If you don’t want to make a rosdep key, but the package is available in your system package manager (eg ``apt``), you can install and use the package that way:

.. code-block:: console

    sudo apt install python3-serial

> If the package is available on The Python Package Index (PyPi) and you want to install globally on your system:

Reviewer: Add link to Python Package Index (https://pypi.org/)

.. code-block::

    python3 -m pip install -U pyserial

If the package is available on PyPi and you want to install locally to your user:

.. code-block:: console

    python3 -m pip install -U --user pyserial

Installing via a virtual environment
------------------------------------

First, create a Colcon workspace:

.. code-block:: console

    mkdir -p ~/colcon_venv/src
    cd ~/colcon_venv/

Then setup your virtual environment:

.. code-block:: console

    # Make a virtual env and activate it
    virtualenv -p python3 ./venv
    source ./venv/bin/activate
    # Make sure that colcon doesn’t try to build the venv
    touch ./venv/COLCON_IGNORE

Next, install the Python packages that you want in your virtual environment:

.. code-block:: console

    python3 -m pip install gtsam pyserial… etc

> Now you can build your workspace and run your python node for great profit.

Reviewer: ```suggestion
Now you can build your workspace and run your python node that depends on packages installed in your virtual environment.
```

Maybe we can drop this part ;)

.. code-block:: console

    # Source Foxy and build
    source /opt/ros/foxy/setup.bash
    colcon build

.. note::

    If you want release your package on Bloom, you should to add the packages you require to ``rosdep``, see the `rosdep key contribution guide`_.

> .. _rosdep key contribution guide: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#rosdep-rules-contributions/

Reviewer: ```suggestion
.. _rosdep key contribution guide: http://docs.ros.org/en/independent/api/rosdep/html/contributing_rules.html
```
